lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -358,6 +361,13 @@ LookupTypeResult TypeChecker::lookupMemberType(DeclContext *dc,
       }
     }
 
+    // Ignore typealiases found in protocol members.
+    if (auto alias = dyn_cast<TypeAliasDecl>(typeDecl)) {
+      auto aliasParent = alias->getParent();
+      if (aliasParent != dc && isa<ProtocolDecl>(aliasParent))
+        continue;
+    }
```

Comment:
I agree that we should normally avoid choosing overloads in this situation. However, that is a more invasive change in the way we handle lookup, so I'm okay with this PR going in as-is.